### PR TITLE
Add blacklist and list of blacklisted phrases/characters

### PR DESCRIPTION
**Pull request by mgejay**
Solves #177

### CODE REVIEW

**Comment by CameronRedmore** [comment]
This seems to be okay, there is no nick blacklist however.

**Comment by Wild1145** [comment]
The blacklist should be stored as part of the settings table in mySQL

**Comment by mgejay** [comment]
@Wild1145 That would create more data for no reason. The blacklist doesn't disappear on a reload or restart like HashMaps and ArrayLists, this is always there and works just fine, no need to create more work and fix something if it isn't broken.

**Comment by Wild1145** [comment]
@Adaptified It does however need to be centrally stored and able to be updated in real time as we need.

**Comment by mgejay** [comment]
@Wild1145 You're developers, it's not hard to edit the code. That's why github is here, people suggest something and then it happens. Not everything needs to be SQL.

**Comment by Wild1145** [comment]
@Adaptified The entire point of this project is that it is fully SQL based, no configigs and everything is global. I and @CameronRedmore may be Java developers, but @TheCJGCJG is not, neither is Varuct, so if they want to make a change, they have to wait for me or Camzie to be free, make the change, test the change and then push and compile. We then have to take the network offline to update it. Or we just follow the project goals to SQL everything and make one minor change to a table and boom, done.

**Comment by mgejay** [comment]
@Wild1145 Done.

**Comment by CameronRedmore** [comment]
The blacklist this takes is simply a comma delimited list in the format: "Admin, Mod, Super, Senior, Telnet, Developer, Manager, Executive, &k, &o, &n, &m, &1, &0" for example. If this string is entered as the value for a new row in the 'settings' table which is named 'blacklist' all of those terms would be blacklisted.

File: src/com/superiornetworks/icarus/ICM_SqlHandler.java [hunk 1]
```diff
@@ -230,6 +230,7 @@ public static boolean isGod(String playerName) throws SQLException
         }
         return false;
     }
+
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/ICM_SqlHandler.java:233]: This should not be there...

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 2]
```diff
@@ -13,77 +13,71 @@
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.entity.Player;
 
-@CommandParameters(name = "nick", description = "Set your nick.", usage = "/<command> <set:playername> <nick>", rank = ICM_Rank.Rank.OP)
-public class Command_nick
-{
+@CommandParameters(name = "nick", description = "Set your nick.", usage = "/<command> <set:playername:blacklist> <nick>", rank = ICM_Rank.Rank.OP)
+public class Command_nick {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 3]
```diff
@@ -13,77 +13,71 @@
...
 
-    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args)
-    {
-        try
-        {
-            if (args.length < 2)
-            {
+    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args) {
+        try {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 4]
```diff
@@ -13,77 +13,71 @@
...
+            if (args.length < 2) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 5]
```diff
@@ -13,77 +13,71 @@
...
                 return false;
             }
 
-            if (args[0].equalsIgnoreCase("set"))
-            {
-                if (sender instanceof ConsoleCommandSender)
-                {
+
+            if (args[0].equalsIgnoreCase("set")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 6]
```diff
@@ -13,77 +13,71 @@
...
+                if (sender instanceof ConsoleCommandSender) {
                     sender.sendMessage(ChatColor.DARK_RED + "You can only set your nick in game.");
                     return true;
                 }
-                String tag = ICM_Utils.buildMessage(args, 1);
+                String nick = ICM_Utils.buildMessage(args, 1);
 
-                if (tag.contains("`") || tag.contains("'"))
-                {
+                if (nick.contains("`") || nick.contains("'")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 7]
```diff
@@ -13,77 +13,71 @@
...
                     sender.sendMessage(ChatColor.DARK_RED + "For security reasons, you cannot use these charecters: ` or '");
                     return true;
                 }
-                if (tag.length() > 25)
-                {
+                if (nick.length() > 25) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 8]
```diff
@@ -13,77 +13,71 @@
...
-                sender.sendMessage(ChatColor.GREEN + "Your nick is now: " + tag);
-            }
-            else
-            {
-
-                if (!ICM_Rank.isRankOrHigher(sender, ICM_Rank.Rank.SUPER))
-                {
+                ICM_SqlHandler.setNickname(sender.getName(), nick);
+                sender.sendMessage(ChatColor.GREEN + "Your nick is now: " + nick);
+            } else {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 9]
```diff
@@ -13,77 +13,71 @@
...
+                if (!ICM_Rank.isRankOrHigher(sender, ICM_Rank.Rank.SUPER)) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 10]
```diff
@@ -13,77 +13,71 @@
...
                     sender.sendMessage(ICM_Utils.NO_PERMS_MESSAGE);
                 }
 
-                if (!ICM_SqlHandler.playerExists(args[0]))
-                {
+                if (!ICM_SqlHandler.playerExists(args[0])) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 11]
```diff
@@ -13,77 +13,71 @@
...
                     sender.sendMessage(ChatColor.DARK_RED + "That player has not joined before.");
                     return true;
                 }
 
-                String tag = ICM_Utils.buildMessage(args, 1);
-                if (tag.contains("`") || tag.contains("'"))
-                {
+                String nick = ICM_Utils.buildMessage(args, 1);
+                if (nick.contains("`") || nick.contains("'")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 12]
```diff
@@ -13,77 +13,71 @@
...
                     sender.sendMessage(ChatColor.DARK_RED + "For security reasons, you cannot use these charecters: ` or '");
                     return true;
                 }
-                if (tag.length() > 25)
-                {
+                if (nick.length() > 25) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 13]
```diff
@@ -13,77 +13,71 @@
...
                     return true;
                 }
-                ICM_SqlHandler.setNickname(args[0], tag);
-                sender.sendMessage(ChatColor.GREEN + "You set that player's nick to: " + tag);
+                for (String blacklist : ((String) ICM_SqlHandler.getFromTable("settingName", "blacklist", "string", "settings")).split(", ")) {
+                    if (nick.toLowerCase().contains(blacklist.toLowerCase()) && !ICM_Rank.isRankOrHigher(sender, ICM_Rank.Rank.SUPER)) {
+                        sender.sendMessage(ChatColor.DARK_RED + "Illegal characters have been detected!\n(If you are unsure what's blacklisted, do /nick blacklist)");
+                        return true;
+                    }
+                    if (args[0].equalsIgnoreCase("blacklist")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 14]
```diff
@@ -13,77 +13,71 @@
...
                 }
-                if (tag.length() > 25)
-                {
+                if (nick.length() > 25) {
                     sender.sendMessage(ChatColor.DARK_RED + "Nicks cannot be larger than 25 charecters.");
                     return true;
                 }
-                ICM_SqlHandler.setNickname(args[0], tag);
-                sender.sendMessage(ChatColor.GREEN + "You set that player's nick to: " + tag);
+                for (String blacklist : ((String) ICM_SqlHandler.getFromTable("settingName", "blacklist", "string", "settings")).split(", ")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 15]
```diff
@@ -13,77 +13,71 @@
...
+                    }
+                }
+
+                ICM_SqlHandler.setNickname(args[0], nick);
+                sender.sendMessage(ChatColor.GREEN + "You set that player's nick to: " + nick);
                 Player p = Bukkit.getServer().getPlayer(args[0]);
-                if (p != null)
-                {
-                    p.sendMessage(ChatColor.GREEN + "Your tag was changed by: " + sender.getName() + " to: " + tag);
+                if (p != null) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 16]
```diff
@@ -13,77 +13,71 @@
...
+                    p.sendMessage(ChatColor.GREEN + "Your nick was changed by: " + sender.getName() + " to: " + nick);
                 }
             }
-        }
-        catch (SQLException ex)
-        {
+        } catch (SQLException ex) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 17]
```diff
@@ -13,10 +13,11 @@
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.entity.Player;
 
-@CommandParameters(name = "tag", description = "Set your tag.", usage = "/<command> <set:playername> <tag>", rank = ICM_Rank.Rank.OP)
+@CommandParameters(name = "tag", description = "Set your tag.", usage = "/<command> <set:playername:blacklist> <tag>", rank = ICM_Rank.Rank.OP)
 public class Command_tag
 {
 
+
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java]: This should not be here.

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 18]
```diff
@@ -25,6 +26,7 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
             {
                 return false;
             }
+
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java]: This should not be here.

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 19]
```diff
@@ -73,6 +75,17 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
                     sender.sendMessage(ChatColor.DARK_RED + "Tags cannot be larger than 25 charecters.");
                     return true;
                 }
+                for (String blacklist : ((String) ICM_SqlHandler.getFromTable("settingName", "blacklist", "string", "settings")).split(", ")) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java]: Formatting Issues

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 20]
```diff
@@ -73,6 +75,17 @@ public boolean onCommand(CommandSender sender, Command cmd, String commandLabel,
...
+                    if (tag.toLowerCase().contains(blacklist.toLowerCase()) && !ICM_Rank.isRankOrHigher(sender, ICM_Rank.Rank.SUPER)) {
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java]: Formatting Issues

**Comment by mgejay** [comment]
Holy formatting issues. I am guessing you want the bracket under the statement.

**Comment by Wild1145** [comment]
@Adaptified It should be in allman style. Just like TFM.

**Comment by mgejay** [comment]
@Wild1145 All done.

**Review by Wild1145 — COMMENTED** [review]
@CameronRedmore Could you take a look and properly review this when you get a moment please.

File: src/com/superiornetworks/icarus/ICM_SqlHandler.java [hunk 21]
```diff
@@ -230,7 +230,7 @@ public static boolean isGod(String playerName) throws SQLException
         }
         return false;
     }
-
+
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/ICM_SqlHandler.java:233]: This shouldnt be here.

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 22]
```diff
@@ -13,80 +13,93 @@
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.entity.Player;
 
-@CommandParameters(name = "nick", description = "Set your nick.", usage = "/<command> <set:playername> <nick>", rank = ICM_Rank.Rank.OP)
-public class Command_nick
+@CommandParameters(name = "nick", description = "Set your nick.", usage = "/<command> <set:playername:blacklist> <nick>", rank = ICM_Rank.Rank.OP)
+public class Command_nick
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java:17]: This shouldnt be here.

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 23]
```diff
@@ -13,80 +13,93 @@
...
+public class Command_nick 
 {
 
-    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args)
-    {
+    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args) {
         try
         {
-            if (args.length < 2)
+            if (args.length < 2)
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java:23]: This shouldnt be here.

File: src/com/superiornetworks/icarus/commands/Command_nick.java [hunk 24]
```diff
@@ -13,80 +13,93 @@
...
                 {
-                    p.sendMessage(ChatColor.GREEN + "Your tag was changed by: " + sender.getName() + " to: " + tag);
+                    p.sendMessage(ChatColor.GREEN + "Your nick was changed by: " + sender.getName() + " to: " + nick);
                 }
             }
-        }
-        catch (SQLException ex)
-        {
+        } 
+            catch (SQLException ex)
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_nick.java:98]: Formatting appears to be out here.

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 25]
```diff
@@ -13,18 +13,18 @@
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.entity.Player;
 
-@CommandParameters(name = "tag", description = "Set your tag.", usage = "/<command> <set:playername> <tag>", rank = ICM_Rank.Rank.OP)
+@CommandParameters(name = "tag", description = "Set your tag.", usage = "/<command> <set:playername:blacklist> <tag>", rank = ICM_Rank.Rank.OP)
 public class Command_tag
 {
-
+
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java:19]: This shouldnt be here.

File: src/com/superiornetworks/icarus/commands/Command_tag.java [hunk 26]
```diff
@@ -13,18 +13,18 @@
...
+  
     public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args)
     {
         try
         {
             if (args.length < 2)
             {
                 return false;
-            }
+            }
```
Comment (Wild1145) [inline src/com/superiornetworks/icarus/commands/Command_tag.java:27]: This shouldnt be here.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
There are some formatting changes that need to be made. You also need to actually create the blacklist in the database upon plugins startup if it is not already created, as right now it does not do that.

**Pull request closed without merging**